src/wallet/components/WalletModal.tsx

```diff
@@ -0,0 +1,267 @@
+        fill="white"
+      />
+    </g>
+  </svg>
+);
+
+interface WalletModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export function WalletModal({ isOpen, onClose }: WalletModalProps) {
+  const { connect } = useConnect();
+  const connectors = useConnectors();
+  const { config } = useOnchainKit();
+
+  const appLogo = config?.appearance?.logo;
+  const appName = config?.appearance?.name;
+  const privacyPolicyUrl =
+    config?.wallet?.privacyUrl ?? 'https://base.org/privacy-policy';
+  const termsOfServiceUrl =
+    config?.wallet?.termsUrl ?? 'https://base.org/terms-of-service';
+
+  return (
+    <div
+      className={cn(
+        `${background.default}/50`,
+        'fixed inset-0 z-50 flex items-start justify-center',
+        !isOpen && 'hidden',
+      )}
+      onClick={() => onClose()}
+    >
+      <div
+        className={cn(
+          border.radius,
+          background.default,
+          line.default,
+          'mt-[116px] h-[390px] w-[323px] p-6 pb-4',
+          'border-x-0 border-t border-b-0',
+          'flex flex-col gap-4',
+        )}
+        onClick={(e) => e.stopPropagation()}
+      >
+        {(appLogo || appName) && (
+          <div className='mt-3 flex h-[116px] w-[275px] flex-col items-center gap-3 self-stretch'>
+            {appLogo && (
+              <div className={cn(border.radius, 'h-14 w-14 overflow-hidden')}>
+                <img
+                  src={appLogo}
+                  alt={`${appName || 'App'} icon`}
+                  className="h-full w-full object-cover"
+                />
+              </div>
+            )}
+            {appName && (
+              <h2
+                className={cn(
+                  text.headline,
+                  color.foreground,
+                  'font-dmSans font-semibold',
+                )}
+              >
+                {appName}
+              </h2>
+            )}
+          </div>
+        )}
+
+        <div className="flex flex-col gap-3">
+          <button
+            onClick={() => {
+              // WalletPreference.SMART_WALLET
+              connect({ connector: connectors[0] });
+              onClose();
+            }}
```

Comment:
can you move this to a function defined above and wrap in `useCallback`

Reply:
Extracting and memoizing all connectors now